apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/utils.ts

```diff
@@ -24,6 +24,14 @@ export const redirectToWpcomPath = ( url: string ) => {
 	redirectParentWindow( `${ origin }${ path }` );
 };
 
+export const getCurrentLaunchFlowUrl = (): string | undefined => {
+	try {
+		return window?.calypsoifyGutenberg?.currentCalypsoUrl as string;
+	} catch ( err ) {
+		return undefined;
+	}
+};
+
```

Comment:
Isn't the Calypso data already available by the time we execute this code inside the iframe? At least this is what I'd assume by looking at `apps/editing-toolkit/editing-toolkit-plugin/editor-gutenboarding-launch/index.ts`.

```suggestion
export const currentLaunchFlowUrl = window?.calypsoifyGutenberg?.currentCalypsoUrl;
```

There is no need to typecast if we add `currentCalypsoUrl?: string;` at line 13 above.

Also I think it's fine to use `window.location.href` as a fallback here in the context of #47803

Reply:
>Isn't the Calypso data already available by the time we execute this code inside the iframe?

The suggested changes don't return the Calypso URL so I wouldn't assume so, but I will revisit this to investigate further.